### ISSUE TITLE
Local variables are treated as properties

### ISSUE THREAD

**Issue by taisuke-fujimoto** [pinterest/ktlint#2779]
## Expected Behavior
This code should not have any lint errors.
```kotlin
package my

import io.kotest.core.spec.style.StringSpec

@Suppress("LocalVariableName")
class SampleTest :
    StringSpec({
        val _test = "test"
        println(_test)
    })

@Suppress("LocalVariableName")
fun test() {
    val _test = "test"
    println(_test)
}
```

## Observed Behavior
This code reports the following lint errors:
```
Sandbox.kt:8:13 Backing property not allowed when 'private' modifier is missing (cannot be auto-corrected)
Sandbox.kt:8:13 Backing property is only allowed when a matching property or function exists (cannot be auto-corrected)
Sandbox.kt:14:9 Backing property not allowed when 'private' modifier is missing (cannot be auto-corrected)
Sandbox.kt:14:9 Backing property is only allowed when a matching property or function exists (cannot be auto-corrected)
```

it seems that local variables are treated as properties.

## Your Environment
* Version of ktlint used: 1.3.1
* Relevant parts of the `.editorconfig` settings : no settings
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): Ktlint Gradle plugin 12.1.1
* Version of Gradle used (if applicable): 8.10
* Operating System and version: Windows 11

**Comment by paul-dingemans** [comment]
> it seems that local variables are treated as properties.

From the perspective of the Kotlin Parser, the local variable is indeed a property:
<img width="908" alt="Screenshot 2024-08-28 at 14 37 24" src="https://github.com/user-attachments/assets/88105ffc-be8b-4575-b8a0-b72a4bcfd901">

It seems that your expectation is that the `@Suppress("LocalVariableName")` would also supress the ktlint violation. This is a small fix which makes sense to me.
<img width="667" alt="Screenshot 2024-08-28 at 14 40 00" src="https://github.com/user-attachments/assets/314f665a-524c-4513-892d-cb50cf4ccc7e">

Until this is implemented, you can suppress the ktlint violation explicitly:
```
@Suppress("LocalVariableName", "ktlint:standard:backing-property-naming")
fun test() {
    val _test = "test"
    println(_test)
}
```
